### PR TITLE
[SYCLomatic] Add support for cudaTextureObject_t that passed into kernel in a struct

### PR DESCRIPTION
**Pull request by ziranzha**

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/test/dpct/texture_object.cu [hunk 1]
```diff
@@ -289,3 +289,46 @@ void foo(){
...
+struct TexList {
+  cudaTextureObject_t tex1;  
+  cudaTextureObject_t tex2;
+  cudaTextureObject_t tex3;
+};
+
+// CHECK: void texlist_device(TexList list, dpct::image_accessor_ext<int, 1> list_tex1) {
+__device__ void texlist_device(TexList list) {
+  int a;
+  tex1Dfetch(&a, list.tex1, 1);
```
Comment (zhimingwang36) [inline clang/test/dpct/texture_object.cu:303]: add checker for Line301.

File: clang/test/dpct/texture_object.cu [hunk 2]
```diff
@@ -289,3 +289,46 @@ void foo(){
...
+}
+
+// CHECK: void texlist_kernel(TexList list, dpct::image_accessor_ext<int, 1> list_tex1,
+// CHECK-NEXT:                     dpct::image_accessor_ext<float, 2> list_tex2,
+// CHECK-NEXT:                    dpct::image_accessor_ext<float, 2> list_tex3) {
+__global__ void texlist_kernel(TexList list) {
+  float b;
+  texlist_device(list);
+  tex2D(&b, list.tex2, 0.5f, 0.5f);
```
Comment (zhimingwang36) [inline clang/test/dpct/texture_object.cu:316]: add check for Line 310 and 311

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**